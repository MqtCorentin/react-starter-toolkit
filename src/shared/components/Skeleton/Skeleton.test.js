import React from 'react';
import { render } from '@testing-library/react';
import SkeletonEnhanced, { Skeleton } from './Skeleton';

describe('<SkeletonEnhanced/>', () => {
  it('Render <SkeletonEnhanced/>', () => {
    const { asFragment } = render(<SkeletonEnhanced />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should contain .af-skeleton', () => {
    const { getByRole, container } = render(<SkeletonEnhanced classModifier="loader" />);
    expect(getByRole('status')).toBeDefined();
    expect(container.querySelector('.af-skeleton--loader')).toBeDefined();
  });
});

describe('<Skeleton/>', () => {
  it('Render <Skeleton/>', () => {
    const { asFragment } = render(<Skeleton />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should contain .af-skeleton', () => {
    const { getByRole, container } = render(<Skeleton className="af-test" />);
    expect(getByRole('status')).toBeDefined();
    expect(container.querySelector('.af-test')).toBeDefined();
  });
});
