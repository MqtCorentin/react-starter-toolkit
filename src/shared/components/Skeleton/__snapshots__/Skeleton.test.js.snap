// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton/> Render <Skeleton/> 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    aria-label="loader"
    class="af-skeleton"
    role="status"
  />
</DocumentFragment>
`;
